core/src/components/radio-group/test/radio-group.e2e.ts

```diff
@@ -0,0 +1,88 @@
+import { newE2EPage } from '@stencil/core/testing';
+
+/**
+ * @param page the E2E page that contains the radio button
+ * @param radioButtonId the id of the radio button to focus
+ * @returns the checked property of the focused radio button
+ */
+const selectRadio = async (page, radioButtonId: string, selectionMethod: 'keyboard' | 'mouse'): Promise<boolean> => {
+  const selector = `ion-radio#${radioButtonId}`;
+  if (selectionMethod === 'keyboard') {
+    await page.focus(selector);
+    await page.keyboard.press('Space');
+  } else if (selectionMethod === 'mouse') {
+    await page.click(selector);
+  }
+  
+  await page.waitForChanges();
+
+  const radioGroup = await page.find(`ion-radio#${radioButtonId} >>> input`);
+  const checked = await radioGroup.getProperty('checked');
+  return checked;
```

Comment:
Any reason why we can't just look at the `checked` property on the host `ion-radio` element instead of diving into the shadow dom?

Reply:
Discussed offline to keep test as is. `checked` is a state not a prop.